pkg/logger/progress_logger.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	pb "github.com/cheggaaa/pb/v3"
+	regv1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// ProgressLogger interface should match with imgpkg util Progress logger
+type ProgressLogger interface {
+	Start(ctx context.Context, progress <-chan regv1.Update)
+	End()
+}
+
+// NewProgressBar constructor to build a ProgressLogger responsible for printing out a progress bar using updates when
+// writing to a registry via ggcr
+func NewProgressBar(writer io.Writer) ProgressLogger {
+	return &ProgressBar{writer: writer}
+}
+
+// NewNoopProgressBar constructs a Noop Progress bar that will not display anything
+func NewNoopProgressBar() ProgressLogger {
+	return &ProgressBarNoTTY{}
+}
+
+// ProgressBarLogger display progress bar on output
+type ProgressBar struct {
+	cancelFunc context.CancelFunc
+	bar        *pb.ProgressBar
+	writer     io.Writer
+}
+
+// Start the display of the Progress Bar
+func (l *ProgressBar) Start(ctx context.Context, progressChan <-chan regv1.Update) {
```

Comment:
add unit tests for `progressBar`  struct. remove tests for noop progress bar

Reply:
Done